pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -53,6 +53,19 @@ def on_trainer_init(
             progress_bar_refresh_rate, process_position
         )
 
+    def resolve_resume_from_checkpoint(self):
+        if not self._trainer_has_checkpoint_callbacks():
+            return self.trainer.resume_from_checkpoint
+        checkpoint_callbacks = self.trainer.checkpoint_callbacks[0]
+        if os.path.exists(checkpoint_callbacks.best_model_path):
+            resume_from_checkpoint_options = [
+                checkpoint_callbacks.best_model_path,
+                self.trainer.resume_from_checkpoint
+            ]
+            resume_from_checkpoint_options.sort()
+            return resume_from_checkpoint_options[-1]
```

Comment:
This solution implies that all checkpoint names will consistently (lexicographically) sort-able.

It works if your checkpoints are named `epoch=N.ckpt` but what if you have something like `best_precision-epoch100.ckpt` and `best_recall=epoch=10.ckpt`. The second one will be used because `"p" < "r" == True`

So I don't think Lightning can make this choice

Reply:
You are right. Let me work on a better solution.